File: java/design-patterns/chain/handler/SupportHandler.java
public abstract class SupportHandler {
    private SupportHandler nextHandler;

    public SupportHandler setNext(SupportHandler handler) {
        this.nextHandler = handler;
        return handler;
    }
    public void handle(Ticket ticket) {
        if (canHandle(ticket)) {
            resolve(ticket);
        } else if (nextHandler != null) {
            nextHandler.handle(ticket);
        } else {
            System.out.println("End of chain reached.");
        }
    }
    protected abstract boolean canHandle(Ticket ticket);
    protected abstract void resolve(Ticket ticket);
}

File: python/design-patterns/chain/handler/support_handler.py
from abc import ABC, abstractmethod

class SupportHandler(ABC):
    def __init__(self):
        self._next_handler = None

    def set_next(self, handler):
        self._next_handler = handler
        return handler

    def handle(self, ticket):
        if self._can_handle(ticket):
            self._resolve(ticket)
        elif self._next_handler:
            self._next_handler.handle(ticket)
        else:
            print("End of chain: unresolved")

    @abstractmethod
    def _can_handle(self, ticket) -> bool: pass
    @abstractmethod
    def _resolve(self, ticket): pass